subiquity/models/filesystem.py

```diff
@@ -47,6 +48,84 @@ class NotFinalPartitionError(Exception):
     the last one."""
 
 
+@attr.s(auto_attribs=True)
+class RecoveryKeyHandler:
+    # Where to store the key on the live system
+    live_location: Optional[pathlib.Path]
+    # Where to store the key in the target system. /target will automatically
+    # be prefixed.
+    backup_location: pathlib.Path
+
+    _key: Optional[str] = attr.ib(repr=False, default=None)
+
+    @classmethod
+    def from_spec(
+        cls, spec: Optional[Dict[str, Any]]
```

Comment:
We do use these "spec" dictionaries for filesystem stuff, but want to move away from them.  Would you create a new attrs object containing your desired fields?